matchers/groovy/todomatcher.go

```diff
@@ -7,10 +7,10 @@ import (
 )
 
 var singleLineTodoPattern = regexp.MustCompile("^\\s*//.*TODO")
-var singleLineValidTodoPattern = regexp.MustCompile("^\\s*// TODO (#?[a-zA-Z0-9\\-]+):.*")
+var singleLineValidTodoPattern = regexp.MustCompile("^\\s*// TODO #?([a-zA-Z0-9\\-]+):.*")
 
 var multiLineTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*TODO")
-var multiLineValidTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*TODO (#?[a-zA-Z0-9\\-]+):.*")
+var multiLineValidTodoPattern = regexp.MustCompile("(?s)^\\s*/\\*.*TODO #?([a-zA-Z0-9\\-]+):.*")
```

Comment:
There's an easier way to achieve this.

See `issuetracker.TaskURLSuffixFor`. In it, you can remove the `#` if it is present. That way, one can specify e.g. `TODO #53`, but the actual request made to the issuetracker will be without the hashtag.

Additionally, this change should be done for github only, because it is natural in it to provide issues with a hashtag, but not as natural in e.g. Jira.

Reply:
I'm sorry that I misunderstood that issue. I'll fix it later